ZCS-1117 JUnit4 + own user

tmclane opened the pull request.

gordyt commented on store/src/java/com/zimbra/qa/unittest/TestDataSource.java:
```diff
@@ -556,14 +579,15 @@ public void tearDown()
         TestUtil.deleteAccount(TEST_USER_NAME);
-        TestUtil.deleteTestData(USER_NAME, NAME_PREFIX);
+        TestUtil.deleteAccount(USER_NAME_2);
+        //TestUtil.deleteTestData(USER_NAME, NAME_PREFIX);
```
nit - remove commented out line

tmclane replied:
Does anyone know if this line is actually necessary?  With it in bad things happen but I wasn't sure if by removing it data was being left behind in the system.

fatkudu commented on store/src/java/com/zimbra/qa/unittest/TestDataSource.java:
```diff
@@ -65,11 +76,15 @@
+    private static String USER_NAME_2;
     private static final String DS_NAME = "TestDataSource";
-    private static final String TEST_USER_NAME = "testdatasource";
+    private static final String TEST_USER_NAME = String.format("%s-%s-testuser1", TestDataSource.class.getSimpleName(), testInfo.getMethodName()).toLowerCase();
```
Can't (meaningfully) use `testInfo.getMethodName()` in a static.
Make this non-final and initialize it in setUp()

tmclane commented:
@fatkudu This has been updated to move the test user string initialization into the setUp method with the others.

tmclane commented:
@fatkudu @grishick @gordyt @ilyusha 
One last update for this ticket for now:

I have moved things around and got the `TestUtil.deleteTestData` call to work again.
Also I added 'reasons' to the assertFail / assertTrue in the `testIsScheduled` test.
Currently this has exposed what I believe to be an actual real failure as shown below:

```<RunUnitTestsResponse numFailed="1" numSkipped="0" numExecuted="6" xmlns="urn:zimbraAdmin">
  <results>
    <completed name="testPollingInterval" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="0.77"/>
    <completed name="testErrorStatus" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="3.62"/>
    <completed name="testMigratePollingInterval" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="0.64"/>
    <completed name="testPop3" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="3.02"/>
    <completed name="testRss" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="1.04"/>
    <failure name="testIsScheduled" class="com.zimbra.qa.unittest.TestDataSource" execSeconds="0.49">DataSource was scheduled when NOT expected - Missing interval</failure>
  </results>
</RunUnitTestsResponse>```

The issue being that a DataSource scheduled with no polling interval is being 'scheduled' when it should not be.

tmclane commented:
Will open a ticket for the failure.
